[NONE] : build health test 준비 pr 통해서 build test 정상 작동 확인 필요

GomHyeok opened the pull request:
## 티켓 링크

## 구현내용
- build health test 준비
- tuist 기반 test 진행 -> 모든 uinit test 통과하도록 작성 (향후 수정 가능)

## 고민사항
- build test가 정상적으로 동작하는지 test필요함

## 기타

GomHyeok committed:
[NONE] : build health test 준비 pr 통해서 build test 정상 작동 확인 필요

TTOzzi commented on .github/workflows/buildTest.yml:
```diff
@@ -0,0 +1,58 @@
+  build-and-test:
+    runs-on: macos-latest
+    env:
+      TUIST_VERSION: 4.12.1
```
요건 `mise.toml` 의 버전값을 참조하도록 해야할 것 같아요.

TTOzzi commented on .github/workflows/buildTest.yml:
```diff
@@ -0,0 +1,58 @@
+      - name: Run Tests
+        working-directory: ${{ steps.detect-tuist-dir.outputs.tuist_path }}
+        run: |
+          tuist test
```
한동안은 테스트가 없을테니, 지금은 `tuist test` 보단 `tuist build` 가 더 적절할 것 같아요.

TTOzzi commented on .github/workflows/buildTest.yml:
```diff
@@ -0,0 +1,58 @@
+          tuist test
+
+      - name: Print Xcode Version
+        run: xcodebuild -version
```
제거해도 될 것 같습니다!

TTOzzi reviewed (commented):
고생하셨습니다! 코멘트 확인 부탁드립니다 🙇

GomHyeok committed:
[REFACTORE] : build.yml 수정
- tuist version -> mise.toml 버전값 참조하도록 변경
- tuist test -> tuist build로 변경
- xcodebuild version 제거

GomHyeok commented:
수정 완료 했습니다!

TTOzzi reviewed (approved):
굿굿 나머진 병합하고 테스트하시죠 👍
